java-util/src/main/java/io/druid/java/util/common/io/smoosh/SmooshedFileMapper.java

```diff
@@ -97,7 +97,7 @@ public static SmooshedFileMapper load(File baseDir) throws IOException
   private final Map<String, Metadata> internalFiles;
   private final List<MappedByteBuffer> buffersList = Lists.newArrayList();
 
-  private SmooshedFileMapper(
+  public SmooshedFileMapper(
```

Comment:
@kaijianding `SmooshedFileMapper` is not a part of public api, and could be changed anytime so that your subclass will break. Worse, the behaviour may break, however without compilation errors, so you won't notice it. So if you want something similar to `SmooshedFileMapper` in your custom SegmentizerFactory, I strongly suggest to copy `SmooshedFileMapper` entirely into your codebase and tailor it for your needs.